### PR TITLE
vimPlugins.CopilotChat-nvim: init at 2024-05-08

### PR DESCRIPTION
**Pull request by MikaelFangel**
## Description of changes

Adds [Copilot Chat](https://github.com/CopilotC-Nvim/CopilotChat.nvim) for nvim.

**EDIT:**
Result of `nixpkgs-review pr 310697` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.CopilotChat-nvim</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
This plugin  depends on `plenary.nvim` and `copilot.lua`. You need to add those dependencies to `vim/plugins/overrides.nix`.

**Comment by MikaelFangel** [comment]
Thank you @GaetanLepage! It's my first time adding a vimPlugin, so hope I've done it right this time (:

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -442,6 +442,10 @@
     dependencies = with self; [ completion-nvim nvim-treesitter ];
   };
 
+  CopilotChat-nvim = super. CopilotChat-nvim.overrideAttrs {
```
Comment (GaetanLepage) [inline pkgs/applications/editors/vim/plugins/overrides.nix:445]:
```suggestion
  CopilotChat-nvim = super.CopilotChat-nvim.overrideAttrs {
```

**Review by GaetanLepage — APPROVED** [review]
You may want to squeeze those two commits into one.
Apart from this, your PR looks perfect to me !
Thanks for the contribution !

**Comment by MikaelFangel** [comment]
Done and thank you for the help with it (:

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310697` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.CopilotChat-nvim</li>
  </ul>
</details>

**Pull request merged**